### PR TITLE
Ensure zlib support in bioconductor-rhdf5.

### PR DESCRIPTION
**Pull request by johanneskoester**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/bioconductor-rhdf5lib/meta.yaml [hunk 1]
```diff
@@ -21,8 +23,12 @@ build:
 requirements:
   host:
     - r-base
+    - zlib
+    - libgfortran
```
Comment (bgruening) [inline recipes/bioconductor-rhdf5lib/meta.yaml]: is `- {{ compiler('fortran') }}` not working?

Comment (johanneskoester) [reply recipes/bioconductor-rhdf5lib/meta.yaml]: Didn't know this is equivalent. I thought it just wants to link against fortran, not building fortran code. But I might be wrong.

File: recipes/bioconductor-rhdf5lib/configure.patch [hunk 2]
```diff
@@ -0,0 +1,11 @@
+--- configure	2018-12-20 17:09:38.001376321 +0100
++++ configure.patched	2018-12-20 17:25:42.834031679 +0100
```
Comment (dpryan79) [inline recipes/bioconductor-rhdf5lib/configure.patch]: You probably want `configure` rather than `configure.patched` there.

**Comment by bgruening** [comment]
@johanneskoester do you know about a test that can ensure zlib support.

**Comment by johanneskoester** [comment]
@bgruening unfortunately not. We could generate an hdf5 file that uses gz compression, but this package, as far as I know, only provides the C library, so it would be some effort to write a test.

File: recipes/bioconductor-rhdf5lib/configure.patch (hunk 2 above)
Comment (johanneskoester) [reply recipes/bioconductor-rhdf5lib/configure.patch]: Isn't that only cosmetic?

**Comment by johanneskoester** [comment]
@dpryan79 I don't think this is a problem in the way conda applies patches. The changes are definitely applied when looking at the logging output.

**Comment by johanneskoester** [comment]
I have added a test case, let's see if it works like that.

**Comment by johanneskoester** [comment]
@dpryan79 @bgruening  the test case works. Unfortunately, it reveals that although we now link rhdf5lib against zlib, there is still a problem opening that file. The requested dataset is definitely gzipped, so there should be no other filter causing a problem. One explanation could be that rhdf5lib still fails to open the right zlib.

**Comment by saketkc** [comment]
Is this good to go @johanneskoester after @enneamer's suggested changes? I just bumped into https://github.com/pachterlab/sleuth/issues/120

**Pull request merged**